### ISSUE TITLE
cert-manager support for spire-controller-manager webhooks

### ISSUE THREAD

**Issue by kfox1111** [spiffe/helm-charts#44]
cert-manager should be an option to get certificates for the spire-controller-manager webhook so they can be standardized over all the webhooks on the cluster.

**Comment by marcofranssen** [comment]
I also prefer to have a single component responsible for my webhooks, though @faisal-memon we might want to file your suggestion as an alternative option for people who don't run cert-manager on their cluster. Could you put that as a separate issue?

**Comment by kfox1111** [comment]
Oh, yeah. Not saying it should be required or even a default to use cert-manager. I just want the option to use it.

**Comment by faisal-memon** [comment]
I think i misunderstood this one, are you saying the webhook for controller-manager?

**Comment by faisal-memon** [comment]
I believe this will need some controller-manager changes.

**Comment by kfox1111** [comment]
Not sure. they seem to have some parts of it in their tree:
https://github.com/spiffe/spire-controller-manager/blob/main/config/crd/patches/cainjection_in_controllermanagerconfigs.yaml

**Comment by kfox1111** [comment]
https://github.com/spiffe/spire-controller-manager/tree/main/config/certmanager

**Comment by kfox1111** [comment]
After reviewing the spire-controller-manager code, I think @faisal-memon is correct. It does look currently impossible to configure it to support cert-manager. I filed https://github.com/spiffe/spire-controller-manager/issues/118 and this issues is blocked on it.
